backend/Social/SocialAPI.py

```diff
@@ -74,15 +74,20 @@ def users():
     try:
         data = db.Profiles.find({"userID": {'$in': userIdList}}, {"_id": 0})
         response = {
-            "data": list(data),
-            "status": "success"
-        }
-        return make_response(response, 400)
+                 "data": list(data),
+                 "status": "success"
+             }
+       
+        if len(userIdList) == 0:
+            return make_response({"err": "userID is not specified", "status": "failed"}), 400 # throws error if userID is not specified in argument
+        for userID in userIdList:
```

Comment:
1, Should use a call with $in to complete the task in 1 query instead of making multiple queries

Reply:
Error only when all userID cannot be found

Reply:
Should use .count() as well if you only need the number of entries, not the data itself

Reply:
Attempt to use ```data.count()``` returns ```'Cursor' object has no attribute 'count'``` error, even though this method is included in [MongoDB documentation](https://docs.mongodb.com/upcoming/reference/method/cursor.count/) (no idea why this is so).

Hence, ```db.Profiles.count_documents({"userID": {'$in': userIdList}}) == 0``` is used as the condition for the non-existence of all userID arguments instead (please refer to latest commit).